testsuite/devserver.go

```diff
@@ -0,0 +1,356 @@
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+// Package testsuite contains unit testing framework for Temporal workflows and activities and a helper to download and
+// start a dev server.
+package testsuite
+
+import (
+	"archive/tar"
+	"archive/zip"
+	"bytes"
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"syscall"
+	"time"
+
+	"go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/internal"
+	ilog "go.temporal.io/sdk/internal/log"
+	"go.temporal.io/sdk/log"
+)
+
+// Cached download of the dev server.
+type CachedDownload struct {
+	// Which version to download, by default the latest version compatible with the SDK will be downloaded.
+	// Acceptable values are specific release versions (e.g v0.3.0), "default", and "latest".
+	Version string
+	// Destination directory or the user temp directory if unset.
+	DestDir string
+}
+
+// Where to find an executable. Can be a path or cached download.
+type EphemeralExe struct {
+	// Existing path on the filesystem for the executable.
+	ExistingPath string
+	// Download the executable if not already there.
+	CachedDownload CachedDownload
+}
+
+// Configuration for the dev server.
+type DevServerOptions struct {
+	// Path to executable or download info - defaults to cached download.
+	Exe EphemeralExe
+	// Client options used to create a client for the dev server.
+	// The provided Namespace or the "default" namespace is automatically registered on startup.
+	// If HostPort is provided, the host and port will be used to bind the server, otherwise the server will bind to
+	// localhost and obtain a free port.
+	ClientOptions *client.Options
+	// SQLite DB filename if persisting or non-persistent if none.
+	DBFilename string
+	// Whether to enable the UI.
+	EnableUI bool
+	// Log format - defaults to "pretty"
+	LogFormat string
+	// Log level - defaults to "warn"
+	LogLevel string
+	// Additional arguments to the dev server.
+	ExtraArgs []string
+}
+
+type DevServer interface {
+	// Stop the running server and wait for shutdown to complete. Error is propagated from server shutdown.
+	Stop() error
+	// Get a connected client, configured to work with the dev server.
+	Client() client.Client
+}
+
+// Temporal CLI based DevServer
+type devServer struct {
+	cmd    *exec.Cmd
+	client client.Client
+}
+
+// StartDevServer starts a Temporal CLI dev server process. This may download the server if not already downloaded.
+func StartDevServer(ctx context.Context, options *DevServerOptions) (DevServer, error) {
```

Comment:
```suggestion
func StartDevServer(ctx context.Context, options DevServerOptions) (DevServer, error) {
```

And just override herein. It's normal for options structs to be by-value for copying (even if they are large).